src/os/bluestore/BlueStore.cc

```diff
@@ -3944,6 +3944,11 @@ bool BlueStore::ExtentMap::encode_some(
 	       << std::dec << " hit new spanning blob " << *p << dendl;
       request_reshard(p->blob_start(), p->blob_end());
       must_reshard = true;
+    } else if (p->blob->is_spanning() && p->logical_end() > end) {
```

Comment:
I have a feeling that resharding doesn't 100% guarantee that spanning blob goes away. It might happen or not depending on shards layout.
Am I missing something?

Reply:
Resharding does not guarantee that spanning blob goes away.
But resharding guarantees that extent will not spill over from one shard to another.